readme.txt
=== cbrnetheme ===

cbrnetheme WordPress Theme

Requires at least: 5.0
Tested up to: 6.3.1
Stable tag: 9.3.5
License: MIT License
License URI: https://opensource.org/licenses/MIT

A starter theme called cbrnetheme.

== Description ==

Air starter theme is built to be very straightforward, front end developer friendly and only partly modular by its structure.

== Installation ==

1. In your admin panel, go to Appearance > Themes and click the Add New button.
2. Click Upload Theme and Choose File, then select the theme's .zip file. Click Install Now.
3. Click Activate to use your new theme right away.

== Frequently Asked Questions ==

= Does this theme support any plugins? =
Yes
== Changelog ==

= 1.0 - 15 Nov 2023 =
* Initial release

== Credits ==
* Based on Air-light by Digitoimisto Dude Oy, Roni Laukkarinen, Copyright (c) 2018 [Digitoimisto Dude Oy](https://www.dude.fi)
* Based on Underscores https://underscores.me/, (C) 2012-2017 Automattic, Inc., [GPLv2 or later](https://www.gnu.org/licenses/gpl-2.0.html)
* normalize.css https://necolas.github.io/normalize.css/, (C) 2012-2016 Nicolas Gallagher and Jonathan Neal, [MIT](https://opensource.org/licenses/MIT)


CHANGELOG.md
### [Unreleased]

* Fix the permission issue with self-updater
* Fix ACF element colors in Gutenberg editor
* Add WOFF2 to variablefont mixin #194 (thanks @raikasdev!)

### 9.3.5: 2023-09-12

* Use semver + date in newtheme.sh start script
* Add self-updater to the start script

### 9.3.4: 2023-09-07

* Fix focus to item when pressing back to top indicator
* Fix Gutenberg admin to use the correct variable font
* Update devpackages to 2.5.4
* Exclude WordPress.DB.SlowDBQuery.slow_db_query_meta_query
* Do not show skip link when focused via mouse from back to top
* Fix external link aria-label getting replaced #193 (kudos to @EliasKau!)
* Improvements and fixes to top scroll button #191 (kudos to @raikasdev!)

### 9.3.3: 2023-05-18

* Reset core-list bullet size
* Remove un-used `image_sizes` array from theme settings
* Run `theme_setup` on `after_setup_theme` hook
* Add custom `pll_translatable` setting to cpt and tax registration, use that to automatically register that content type for Polylang
* Fix ACF button color
* Remove deprecated number-leading-zero rule
* Add support for navigation version where main item is a clickable <button> (kudos to @Tumppex)
* If nav link anchor link, add class js-trigger for moveTo
* Fix JS error with link label not existing
* Close navigation if anchor item is nav-link
* Fix focus to target in anchors.js
* Bump tested up WordPress version to 6.2.1

### 9.3.2: 2023-03-21

* When page is loaded with "s" in url parameters try to set focus to search input field (kudos to @EliasKau)
* Calculate mobile nav top and height in js when there are air notificans present (kudos to @Tumppex)
* Fix irregularities with keyboard navigation #175 (kudos to @michaelbourne!)
* Fix navigation focus trap #175 (kudos to @michaelbourne!)
* Stylelint: Add number-leading-zero from devpackages
* Fix navigation issues #177 (kudos to @michaelbourne!)

### 9.3.1: 2023-02-24

* Remove nav-toggle styles not in use
* Add air-notification support to navigation.js
* Simplify button component even more by removing unused arrow conditional

### 9.3.0: 2023-02-17

* Change nav toggle position to static by default (kudos to @Tumppex)
* Add wrapper to navigation (kudos to @Tumppex)
* Mobile nav variable to change nav width (kudos to @Tumppex)

### 9.2.9: 2023-02-15

* Start script: Copy .nvmrc to project root from devpackages
* Improve block cache bypass logic and show bypass reason on log
* Try to determine if block has Gravity Forms form and always bypass cache if 
* Stylelint fixes
* Simplify button SCSS component
* Remove Travis
* Add GitHub workflows for styles, html, php and js
* Fix version in .nvmrc
* Add badges for GitHub workflow build statuses
* Fix stylelint errors for v15
* Add smaller font-sizes 12 and 13
* Replace all suitable px sizes with rem as per stylelint-rem-over-px
* Update .stylelintrc from devpackages 2.5.3

### 9.2.8: 2023-01-30

* Add gap for categories
* Change default mobile breakpoint to 600px
* Remove breakpoints not in use
* Fix iPad navigation overlapping blocks when closed

### 9.2.7: 2023-01-06

* Remove unused no-js styles
* Remove unused CSS variables
* Simplify defining CSS variables in single use cases
* Remove `<kbd>` styles that are rarely needed
* Fix entry-footer
* Simplify category styles
* Simplify tag styles
* Remove _blog.scss
* Add views/_single.scss
* Use list element in categories instead of a paragraph

### 9.2.6: 2022-12-31

* Fix undefined body class
* Fix nav styles leaking to pagination
* Drop sanitize.css dependency
* Use sanitize.css/reduce-motion baked-in instead from npm

### 9.2.5: 2022-12-31

* Major navigation.js rewrite
* Modularize navigation.js
* Make navigation dependency free, completely dropping jQuery
* Perfected arrow and keyboard navigation patterns for A11y
* Fix aria-labels and aria-expanded
* Move nav-toggle inside nav landmark
* Fix focus trap
* Rewrite nav-toggle and 🍔
* Use clever way to visually present attr(aria-label) for 🍔
* Define mobile breakpoint more modern way with getPropertyValue
* Drop IE10-IE11 support for navigation.js
* Drop hamburgers npm module dependency
* Fix focusable elements if last item is a button
* Fix checkForSubmenuOverflow() function
* Fix resize functions
* Remove useless vars and functions
* Add plus and minus icons for mobile as default as inlined SVG
* Drop devDependencies from theme package.json, add instructions on contributing
* Fix acf-button color inside a Gutenberg block
* Navigation.js: Bail if navigation doesn't exist
* Remove leftover deprecated gutenberg-helpers.js
* Allow :not notations where needed

### 9.2.4: 2022-12-02

* Upgrade to [devpackages 2.4.9](https://github.com/digitoimistodude/devpackages/releases/tag/2.4.9)
* Disable no-descending-specificity rule in stylelint
* Ignore some obvious phpcs nags

### 9.2.3: 2022-11-28

* Add core/list-item to allowed blocks due to WP 6.1 and latest Gutenberg editor
* Remove extraneous padding in the first block on Gutenberg editor
* Make heading CSS variable naming more consistent
* Button component: Do not use border-radius on ghost button as it's inherited from button
* Button component: Use own border radius instead of shared with form fields
* Update generated README.md in the start script
* Fix ACF relation field color in Gutenberg blocks
* Simplify _font-family.scss
* Deprecate core/code, core/columns, core/cover, core/embed, core/preformatted and code/verse CSS from cbrnetheme, use Gutenberg official styles instead
* Add Inter as variable font

### 9.2.2: 2022-11-04

* Fix incorrect textdomain in external-link JS module
* Simplify CPT and taxonomy registration by using the class name also as a slug (Merge pull request #158 from digitoimistodude/cpt-tax-register-simplification, kudos to @timiwahalahti)
* Remove what-input depenency, use :focus and :focus-within instead, remove forced focus outline-color
* Bump tested up WordPress version to 6.1

### 9.2.1: 2022-10-28


functions.php
<?php
/**
 * Gather all bits and pieces together.
 * If you end up having multiple post types, taxonomies,
 * hooks and functions - please split those to their
 * own files under /inc and just require here.
 *
 * @Date: 2019-10-15 12:30:02
 * @Last Modified by:   Roni Laukkarinen
 * @Last Modified time: 2023-09-12 12:20:49
 *
 * @package cbrnetheme
 */

namespace Cbrne_Theme;

/**
 * The current version of the theme.
 */
define('Cbrne_Theme_VERSION', '0.0.1');

// We need to have some defaults as comments or empties so let's allow this:
// phpcs:disable Squiz.Commenting.InlineComment.SpacingBefore, WordPress.Arrays.ArrayDeclarationSpacing.SpaceInEmptyArray

/**
 * Theme settings
 */
add_action('after_setup_theme', function () {
    $theme_settings = [
    /**
     * Theme textdomain
     */
    'textdomain' => 'cbrnetheme',

    /**
     * Content width
     */
    'content_width' => 800,

    /**
     * Logo and featured image
     */
    'default_featured_image'  => null,
    'logo'                    => '/svg/logo.svg',

    /**
     * Custom setting group settings when using Air setting groups plugin.
     * On multilingual sites using Polylang, translations are handled automatically.
     */
    'custom_settings' => [
      // 'your-custom-setting' => [
      //   'id' => Your custom setting post id,
      //   'title' => 'Your custom setting',
      //   'block-editor' => true,
      //  ],
    ],

    'social_media_accounts'  => [
      // 'twitter' => [
      //   'title' => 'Twitter',
      //   'url'   => 'https://twitter.com/digitoimistodude',
      // ],
    ],

    /**
     * All links are cheked with JS, if those direct to external site and if,
     * indicator of that is included. Exclude domains from that check in this array.
     */
    'external_link_domains_exclude' => [
      'localhost:3000',
      'airdev.test',
      'airwptheme.com',
      'localhost',
    ],

    /**
     * Menu locations
     */
    'menu_locations' => [
      'primary' => __('Primary Menu', 'cbrnetheme'),
      'footer' => __('Footer Menu', 'cbrnetheme'),
    ],

    /**
     * Taxonomies
     *
     * See the instructions:
     * https://github.com/digitoimistodude/cbrnetheme#custom-taxonomies
     */
    'taxonomies' => [
      // 'Your_Taxonomy' => [ 'post', 'page' ],
    ],

    /**
     * Post types
     *
     * See the instructions:
     * https://github.com/digitoimistodude/cbrnetheme#custom-post-types
     */
    'post_types' => [
      // 'Your_Post_Type',
    ],

    /**
     * Gutenberg -related settings
     */
    // Register custom ACF Blocks
    'acf_blocks' => [
      // [
      //   'name'           => 'block-file-slug',
      //   'title'          => 'Block Visible Name',
      //   // You can safely remove lines below if you find no use for them
      //   'prevent_cache'  => false, // Defaults to false,
      //   // Icon defaults to svg file inside svg/block-icons named after the block name,
      //   // eg. svg/block-icons/block-file-slug.svg
      //   //
      //   // Icon setting defines the dashicon equivalent: https://developer.wordpress.org/resource/dashicons/#block-default
      //   // 'icon'  => 'block-default',
      // ],
    ],

    // Custom ACF block default settings
    'acf_block_defaults' => [
      'category'          => 'cbrnetheme',
      'mode'              => 'auto',
      'align'             => 'full',
      'post_types'        => [
        'page',
      ],
      'supports'  => [
        'align'           => false,
        'anchor'          => true,
        'customClassName' => false,
      ],
      'render_callback'   => __NAMESPACE__ . '\render_acf_block',
    ],

    // Restrict to only selected blocks
    // Set the value to 'all' to allow all blocks everywhere
    'allowed_blocks' =>[
      'default' => [
      ],
      'post' => [
        'core/archives',
        'core/audio',
        'core/buttons',
        'core/categories',
        'core/code',
        'core/column',
        'core/columns',
        'core/coverImage',
        'core/embed',
        'core/file',
        'core/freeform',
        'core/gallery',
        'core/heading',
        'core/html',
        'core/image',
        'core/latestComments',
        'core/latestPosts',
        'core/list',
        'core/list-item',
        'core/more',
        'core/nextpage',
        'core/paragraph',
        'core/preformatted',
        'core/pullquote',
        'core/quote',
        'core/block',
        'core/separator',
        'core/shortcode',
        'core/spacer',
        'core/subhead',
        'core/table',
        'core/textColumns',
        'core/verse',
        'core/video',
      ],
      'page' => 'all',
    ],

    // If you want to use classic editor somewhere, define it here
    'use_classic_editor' => [],

    // Add your own settings and use them wherever you need, for example THEME_SETTINGS['my_custom_setting']
    'my_custom_cbrne_radio_setting_id' => 'normal',
    ];

    $theme_settings = apply_filters('Cbrne_Theme_theme_settings', $theme_settings);

    define('THEME_SETTINGS', $theme_settings);
}); // end action after_setup_theme

/**
 * Required files
 */
require get_theme_file_path('/inc/hooks.php');
require get_theme_file_path('/inc/includes.php');
require get_theme_file_path('/inc/template-tags.php');

// Run theme setup
add_action('after_setup_theme', __NAMESPACE__ . '\theme_setup');
add_action('after_setup_theme', __NAMESPACE__ . '\build_theme_support');

// REMOVE WP EMOJI
remove_action('wp_head', 'print_emoji_detection_script', 7);
remove_action('wp_print_styles', 'print_emoji_styles');

remove_action('admin_print_scripts', 'print_emoji_detection_script');
remove_action('admin_print_styles', 'print_emoji_styles');
